providers/core/core.battery.js

```diff
@@ -0,0 +1,78 @@
+/*globals navigator*/
+/*jshint node:true*/
+var PromiseCompat = require('es6-promise').Promise;
+var nav = navigator;
+
+var BatteryProvider = function(cap, dispatchEvent) {
+  "use strict";
+  this._batteryPromise = nav && nav.getBattery ? nav.getBattery() : null;
+
+  if (this._batteryPromise) {
+    this._batteryPromise.then(function(battery) {
+      this._setupListeners(battery, dispatchEvent);
+    }.bind(this));
+  }
```

Comment:
Maybe worth adding an `else { console.warn("Trying to use core.battery API without client support"); }` or such.